meson.build

```diff
@@ -640,8 +640,88 @@ conf.set('CONTAINER_UID_BASE_MAX', container_uid_base_max)
 substs.set('containeruidbasemin', container_uid_base_min)
 substs.set('containeruidbasemax', container_uid_base_max)
 
-conf.set_quoted('NOBODY_USER_NAME', get_option('nobody-user'))
-conf.set_quoted('NOBODY_GROUP_NAME', get_option('nobody-group'))
+nobody_user_names = ['nobody', 'nfsnobody']
+nobody_group_names = ['nobody', 'nfsnobody', 'nogroup']
+
+nobody_user_name = get_option('nobody-user')
+# First strategy: check if any one the names we like exists and does what we
+# want. If so, use it. Do all this in the order of preference. Note that we
+# explicitly check the UID assigned to these names, since Fedora for example
+# knows a nobody user with UID 99, and we want to stay away from that.
+foreach name : nobody_user_names
+         if nobody_user_name == ''
+                id_result = run_command('id', '-u', name)
+                if id_result.returncode() == 0 and id_result.stdout().to_int() == 65534
+                        nobody_user_name = name
+                endif
+         endif
+endforeach
+# Second strategy: if none of the names we like exists, see what name is
+# assigned to 65534, if any. Apply one restriction: if the name is from our
+# preferred name list, then NSS lookups are asynmmetric, let's avoid things
```

Comment:
Typo.

Reply:
Hmm, do we really want to go to this trouble? We had an option to specify the group name, and I think that was OK. In the long run, I think we're better of standarizing on the same name.

We'd should file a Fedora Change to drop the one with uid=99, I can do that.

Reply:
the reason i added this was that i ran into trouble with the tests here, since just building systemd on my fedora, installing it and then running the tests again failed, because the install resulted in an asymmetry: resolving "nobody" would yield 99, but resolving 65534 would yield "nobody"... 

So I figured we should make building systemd just work, and figure out the right values, to avoid such testing borkages.

And yeah, we should really fix Fedora. But I figure that's gonna be non-trivial... I started the discussion about that a year back, but never followed through. i figure we have to:

 1. add a transition script that renames the existing use `nobody` (the one with UID 99) to `oldnobody`, and similar for the group. (just to make sure that any potential files continue to be owned by some user, so that UID 99 is not reused otherwise)
 2. stop creating any user/group for 99 on new installs
 3. remove the "nfsnobody" user and group while upgrading
 4. create the new "nobody" user and group as 65534 on old and new installs, from some base system package
 5. fix the "nfsutils" package to not create nfsnobody anymore
 6. audit the nfsutils codebase (possibly more) for code references to nfsnobody, and change it to "nobody"
 7. recompile systemd to use "nobody" now, instead of "nfsnobody"
 8. update the relevant packaging/distro/guidelines docs

And I figure steps 3+4+5+6+7 need to be done in a synchronized upload....

Reply:
oh, and then there's the ugliness that some distros call the group matching "nobody" "nogroup", and others do not... And it's hard to change those too, since it's debian...

Reply:
Hmm, on my F27 system this results in `nobody` being picked, and `nobody` has an entry with uid=99 in /etc/passwd. This just propagates existing breakage.

Frankly, I'd prefer to
a) keep the nobody-user option with `nobody` default
b) check if the user we get from a) resolved to the expected uid and error out if it does not.
This has the advantage that it'll force people who have "broken" systems, as e.g. Fedora, to take notice.